model/labels/labels_string.go

```diff
@@ -423,27 +527,22 @@ func FromStrings(ss ...string) Labels {
 // Compare compares the two label sets.
 // The result will be 0 if a==b, <0 if a < b, and >0 if a > b.
 // TODO: replace with Less function - Compare is never needed.
-// TODO: just compare the underlying strings when we don't need alphanumeric sorting.
 func Compare(a, b Labels) int {
-	l := len(a.data)
-	if len(b.data) < l {
-		l = len(b.data)
-	}
-
+	la, lb := len(a.data), len(b.data)
 	ia, ib := 0, 0
-	for ia < l {
+	for ia < la && ib < lb {
 		var aName, bName string
-		aName, ia = decodeString(a.data, ia)
-		bName, ib = decodeString(b.data, ib)
+		aName, ia = decodeString(a.syms, a.data, ia)
```

Comment:
If the two symbol tables are equal, would it be possible to avoid decoding label symbols and compare them directly?

Reply:
For `Equal`, yes, but not `Compare` since we don’t have an ordering in the symbol table. 
Might be worth skipping all the strings that are equal and just decoding the first one that’s different.

Reply:
That's a good point, decoding only what is different would probably speed things up when doing a merge of two sorted label sets. For example, when merging data from two TSDBs.

Reply:
Similar idea for current main: #12451